### DISCUSSION TITLE
How do i make another bot use my bot commands?

### DISCUSSION THREAD

**Discussion by stephanto-dev** [discord-jda/JDA#1603] [Questions and Help]
Hello, I have a bot made in JS and I want it to be able to use commands from my bot in Java. In JS I know how to allow other bots to use commands in it, but I don't know how to do the same in Java using JDA. Does anyone have any ideas?

**Comment by DV8FromTheWorld** [comment]
Bots can use you commands (Not [Slash Commands](https://discord.com/developers/docs/interactions/slash-commands)!) by default. You have to explicitly check in your event handler code to _disable_ other bots executing your text commands.

[Example of ignoring a bot's message that might invoke your command](https://github.com/DV8FromTheWorld/JDA/blob/development/src/examples/java/AudioEchoExample.java#L72-L74).

**Reply by stephanto-dev** [reply]
So an example, I have a bot that executes a command, in JS, called "!test". After it performs its function, the bot sends a message in the chat written "!test2", in Java. However the second command does not works (I already checked and the problem is not the JS code). Do you believe that it is some Java code that is blocking the bot from executing another command?

**Reply by DV8FromTheWorld** [reply]
You should post your code.

However, yes, you likely have something in your Java code that is explicitly preventing execution. There is nothing to "enable" in JDA to allow execution of commands by other bots.
